kong/api/routes/ca_certificates.lua

```diff
@@ -0,0 +1,56 @@
+local ngx = ngx
+local kong = kong
+local ipairs = ipairs
+local null = ngx.null
+local string_format = string.format
+
+
+local function each_service(entity)
+  local options = {
+    workspace = null,
+  }
+
+  local iter = entity:each(1000, options)
+  local function iterator()
+    local element, err = iter()
+    if err then return nil, err end
+    if element == nil then return end
+    if element.ca_certificates and #element.ca_certificates > 0 then return element, nil end
+    return iterator()
+  end
+
+  return iterator
+end
+
+
+local function verify_references(id)
+  -- services
+  for service, err in each_service(kong.db.services) do
+    if err then
+      ngx.log(ngx.ERR, "could not load services: ", err)
+      return
+    end
+
+    for _, v in ipairs(service.ca_certificates) do
```

Comment:
This seems to be a very hacky solution, as it does not work with anything else that might also reference `ca_certificate` by UUID. I wonder how useful such specific solution is.

Reply:
I agree. Things like mtls-auth plugins will also reference `ca_certificate`. A general solution may require support of entity back-reference mechanism which I think is a big feature.

Reply:
Indeed, this is not an elegant solution, and if new references are added in the future, it is up to the developer to add the corresponding filter detection.